components/collection/CollectionInfo.vue

```diff
@@ -63,18 +67,41 @@ import {
 const route = useRoute()
 const { urlPrefix } = usePrefix()
 const { availableChains } = useChain()
-
 const collectionId = computed(() => route.params.id)
 const chain = computed(
   () =>
     availableChains.value.find((chain) => chain.value === route.params.prefix)
       .text
 )
 const address = computed(() => collectionInfo.value?.currentOwner)
-
+const seeAllDescription = ref(false)
+const DESCRIPTION_MAX_LENGTH = 210
 const { collection: collectionInfo } = useCollectionMinimal({
   collectionId: collectionId.value,
 })
+
+const toggleSeeAllDescription = () => {
+  seeAllDescription.value = !seeAllDescription.value
+}
+
+const hasSeeAllDescriptionOption = computed(() => {
+  return (
+    (collectionInfo.value?.meta?.description?.length || 0) >
+    DESCRIPTION_MAX_LENGTH
+  )
+})
+
+const visibleDescription = computed(() => {
+  const desc = collectionInfo.value?.meta?.description
+
+  return (
+    (!hasSeeAllDescriptionOption.value || seeAllDescription.value
+      ? desc
+      : desc?.slice(0, DESCRIPTION_MAX_LENGTH)
+    )?.replaceAll('\n', '  \n') || ''
+  )
+})
+
 const { stats } = useCollectionDetails({ collectionId: collectionId.value })
```

Comment:
, Would move

It into the component

Reply:
50 via max Len prop it would se available else where

Reply:
Do you mean moving these parts into markdown component? If so, PR #5366 should be merged firs, then we could do the refactor.

Reply:
Lets do it in other PR